File: elixir/delete-nodes-from-linked-list-present-in-array.ex
# Definition for singly-linked list.
#
# defmodule ListNode do
#   @type t :: %__MODULE__{
#           val: integer,
#           next: ListNode.t() | nil
#         }
#   defstruct val: 0, next: nil
# end

defmodule Solution do
  @spec modified_list(nums :: [integer], head :: ListNode.t | nil) :: ListNode.t | nil
  def modified_list(nums, head), do: recur(MapSet.new(nums), head)

  def recur(_, nil), do: nil
  def recur(to_delete, %{val: val, next: next}) do
    if MapSet.member?(to_delete, val) do
      recur(to_delete, next)
    else
      %ListNode{val: val, next: recur(to_delete, next)}
    end
  end
end

File: python3/delete-nodes-from-linked-list-present-in-array.py
from typing import List, Optional


# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        to_remove = set(nums)

        result_ref = ListNode(0)
        res = result_ref
        it = head
        while it is not None:
            if it.val not in to_remove:
                res.next = ListNode(it.val)
                res = res.next
            it = it.next
        return result_ref.next
